codec/src/test/java/io/netty/handler/codec/DatagramPacketEncoderTest.java

```diff
@@ -82,4 +86,39 @@ public void testUnmatchedType() {
         assertTrue(channel.writeOutbound(netty));
         assertSame(netty, channel.readOutbound());
     }
+
+    @Test
+    public void testNotSharable() {
+        DatagramPacketEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>> encoder =
+                new DatagramPacketEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>>(
+                        new MessageToMessageEncoder<AddressedEnvelope<ByteBuf, InetSocketAddress>>() {
+                    @Override
+                    protected void encode(
+                            ChannelHandlerContext ctx, AddressedEnvelope<ByteBuf,
+                            InetSocketAddress> msg, List<Object> out) {
+                            // NOOP
+                    }
+                });
+        assertFalse(encoder.isSharable());
```

Comment:
nit: should we assert that this value equals the value of the `MessageToMessageEncoder#isSharable` ?